lib/puppet/provider/package/snap.rb

```diff
@@ -62,54 +53,59 @@ def latest
   end
 
   def uninstall
-    self.class.modify_snap('remove', @resource[:name])
+    modify_snap('remove')
   end
 
   # Purge differs from remove as it doesn't save snapshot with snap's data.
   def purge
-    self.class.modify_snap('remove', @resource[:name], ['purge'])
+    modify_snap('remove', ['purge'])
   end
 
-  def self.installed_snaps
-    res = PuppetX::Snap::API.get('/v2/snaps')
-
-    raise Puppet::Error, "Could not find installed snaps (code: #{res['status-code']})" unless [200, 404].include?(res['status-code'])
+  def modify_snap(action, options = nil)
+    req = self.class.generate_request(action, determine_channel(options), options)
+    response = PuppetX::Snap::API.post("/v2/snaps/#{@resource[:name]}", req)
+    change_id = PuppetX::Snap::API.get_id_from_async_req(response)
+    PuppetX::Snap::API.complete(change_id)
+  end
 
-    res['result'].map { |hash| hash.slice('name', 'version') } if res['status-code'] == 200
+  def determine_channel(options = nil)
+    channel = self.class.parse_version_from_ensure(@resource[:ensure])
+    channel ||= self.class.parse_channel(options) if options
+    channel ||= 'latest/stable'
+    channel
   end
 
-  def self.generate_request(action, options)
+  def self.generate_request(action, channel, options)
     request = { 'action' => action }
+    request['channel'] = channel unless channel.nil?
```

Comment:
This seems to be a bug of the old implementation as well.
`channel` field is only applicable for actions `install` or `refresh`.